coap/api/transport.go

```diff
@@ -154,8 +155,8 @@ func decodeMessage(msg *mux.Message) (messaging.Message, error) {
 
 func parseID(path string) string {
 	vars := strings.Split(path, "/")
-	if len(vars) > 1 {
-		return vars[1]
+	if len(vars) > 2 {
```

Comment:
It would be good to leave a small comment here, explaining where the number `2` comes from so that the reader can understand quickly.